### PR TITLE
feat: #217 Add runtime validation to service layer (all methods)

### PR DESCRIPTION
**Pull request by Howl404**
Converted only DayReportDTO, if its ok will convert everything else too.

### CODE REVIEW

**Review by tritonJS826 — COMMENTED** [review]
Good, you can add it to every service method! just one question - what will happened if there will be invalid object?  will zod throw an Error in that case? 

By my opinion if yes - we should catch this error globally and handle it, maybe not in the current PR. Shoud I create new task for this?

**Review by tritonJS826 — APPROVED** [review]

**Comment by Howl404** [comment]
@tritonJS826 Yea, Zod throws error with details. I will create separate issue because right now i don't know how to implement global error catcher.

File: mw-webapp/src/service/WayService.ts [hunk 1]
```diff
@@ -67,9 +75,11 @@ export class WayService {
     const waysRef = collection(db, PATH_TO_WAYS_COLLECTION);
     const ownWaysQuery = query(waysRef, where("ownerUuid", "==", uuid));
     const ownWaysRaw = await getDocs(ownWaysQuery);
-    const ownWays = querySnapshotToDTOConverter<WayDTO>(ownWaysRaw);
+    const ownWaysDTO = querySnapshotToDTOConverter<WayDTO>(ownWaysRaw);
 
-    return ownWays;
+    const validatedOwnWaysDTO = WayDTOArraySchema.parse(ownWaysDTO);
+
+    return validatedOwnWaysDTO;
```
Comment (Howl404) [inline mw-webapp/src/service/WayService.ts:82]: If User doesnt have Own\Mentoring\Favorite ways, what should we do? It will throw error every time

**Review by Ekaterina1994 — APPROVED** [review]

**Review by tritonJS826 — APPROVED** [review]

File: mw-webapp/src/model/DTOModel/WayDTO.ts [hunk 2]
```diff
@@ -1,57 +1,52 @@
...
-    this.ownerUuid = wayData.ownerUuid;
-    this.monthReportUuids = wayData.monthReportUuids;
-    this.goalUuid = wayData.goalUuid;
-    this.currentMentorUuids = wayData.currentMentorUuids;
-    this.isCompleted = wayData.isCompleted;
-  }
+}).strict();
 
-}
+export const WayDTOArraySchema = z.array(WayDTOSchema);
```
Comment (tritonJS826) [inline mw-webapp/src/model/DTOModel/WayDTO.ts:47]: Due to the fact that Array is a typescript type, it is usually preferable to use the word "List" in the names or simply add the letter "s", for example "WaysDTO"

**Review by tritonJS826 — APPROVED** [review]

File: mw-webapp/src/service/WayService.ts (hunk 1 above)
Comment (tritonJS826) [reply mw-webapp/src/service/WayService.ts:82]: Why should it throw errors? It should be just an empty array.

Comment (Howl404) [reply mw-webapp/src/service/WayService.ts:82]: My bad, I misunderstood the errors, we should clean our DB a little bit, few ways was without name and other properties

**Pull request merged**